_assets/js/laws-regs-parser.js

```diff
@@ -1,51 +1,79 @@
 export default function parseLawsAndRegs (mainContent) {
     if (!mainContent) return;
 
-    const headings = ['h2', 'h3', 'h4', 'h5', 'strong'];
     const parent = mainContent.parentElement;
-    const newMainContent = document.createElement('div');
-    newMainContent.className = 'interactive-headers';
     const clone = mainContent.cloneNode(true);
     const contentNodes = Array.from(clone.childNodes);
-    const sections = [];
+    const subpartIndices = [0];
+    const subparts = [createDiv('subpart')];
     let lastHeading = 0;
-
-    contentNodes.map((node, i) => {
-        if (!(headings.includes(node.localName)
-            || i === contentNodes.length - 1)) return;
+    let length = 1;
+    const sections = contentNodes.map((node, i) => {
+        const isInteractiveHeader = getIsInteractiveHeader(contentNodes, node, i);
+        if (!isInteractiveHeader) return null;
+        if (node.localName === 'h2') {
+            const subpart = createDiv('subpart');
+            subparts.push(subpart);
+            subpartIndices.push(length);
+        }
         const sectionText = contentNodes
             .slice(lastHeading, i)
             .map(node => node.outerHTML).join(' ');
-        sections.push(sectionText);
         lastHeading = i;
-    });
+        length++
+        const section = createDiv('section');
+        section.innerHTML = sectionText.trim();
+        return section;
+    }).filter(section => !!section);
 
+    let currentSubpart = subparts[0];
     sections.map((section, i) => {
```

Comment:
nit: prefer `forEach` when not using the return value of `map` (here and elsewhere)